tests/test_integrals.py

```diff
@@ -667,38 +695,334 @@ def test_singular_integral_vac_estell(self, interpolator):
         interp = interpolator(grid, grid, st, sz, q)
         Bplasma = virtual_casing_biot_savart(data, data, interp, chunk_size=50)
         # need extra factor of B/2 bc we're evaluating on plasma surface
         Bplasma += data["B"] / 2
         Bplasma = np.linalg.norm(Bplasma, axis=-1)
         # scale by total field magnitude
         B = Bplasma / np.linalg.norm(data["B"], axis=-1).mean()
-        np.testing.assert_allclose(B, 0, atol=0.005)
+        np.testing.assert_allclose(B, 0, atol=0.0054)
 
     @pytest.mark.unit
-    @pytest.mark.parametrize("interpolator", [FFTInterpolator, DFTInterpolator])
-    def test_biest_interpolators(self, interpolator):
-        """Test that FFT and DFT interpolation gives same result for standard grids."""
-        grid = LinearGrid(0, *_f_2d_nyquist_freq())
-        h_t = 2 * np.pi / grid.num_theta
-        h_z = 2 * np.pi / grid.num_zeta / grid.NFP
+    def test_laplace_harmonic_simple(self, chunk_size=2000, resolution=50, atol=1e-4):
+        """Test that Laplace solution recovers expected analytic result.
+
+        Define boundary R_b(θ,ζ) = R₀ + a cos θ and Z_b(θ,ζ) = -a sin θ.
+        θ = 0 is outboard side and θ increases clockwise.
+        Define harmonic map Φ: ρ,θ,ζ ↦ Z(ρ,θ,ζ).
+        Choose b.c. B₀⋅n = -∇ϕ⋅n
+                         = -[0, 0, 1]⋅[cos(θ)cos(ζ), cos(θ)sin(ζ), -sin(θ)] = sin(θ)
+        and test that ‖ Φ − Z ‖_∞ → 0.
+        """
+        a = 1
+        eq = Equilibrium()  # Choosing a = 1.
+        source_grid = LinearGrid(M=resolution, N=resolution, NFP=eq.NFP)
+
+        source_data = eq.compute(["Z", "n_rho", "theta"], grid=source_grid)
+        np.testing.assert_allclose(source_data["Z"], -a * np.sin(source_data["theta"]))
+        B0n = np.sin(source_data["theta"])
+        np.testing.assert_allclose(-source_data["n_rho"][:, 2], B0n, atol=1e-12)
+
+        Phi_mn, Phi_transform = compute_Phi_mn(
+            eq=eq,
+            B0n=B0n,
+            Phi_grid=LinearGrid(M=1, N=0, NFP=eq.NFP),
+            source_grid=source_grid,
+            chunk_size=chunk_size,
+            warn_fft=False,
+        )
+        eval_grid = LinearGrid(M=5, N=5, NFP=eq.NFP)
+        eval_data = eq.compute(["Z", "n_rho", "theta"], grid=eval_grid)
+        eval_data["Phi"] = Transform(eval_grid, Phi_transform.basis).transform(Phi_mn)
+        err = eval_data["Phi"] - eval_data["Z"]
+        # ``atol`` may tend to 0 as ``resolution`` grows
+        np.testing.assert_allclose(np.max(err) - np.min(err), 0, atol=atol)
+
+        B0n = np.sin(eval_data["theta"])
+        dPhi_dn = compute_dPhi_dn(
+            eq=eq,
+            eval_grid=eval_grid,
+            source_grid=source_grid,
+            Phi_mn=Phi_mn,
+            basis=Phi_transform.basis,
+            chunk_size=chunk_size,
+            warn_fft=False,
+        )
+        Bn = B0n + dPhi_dn
+        # FIXME: This fails to converge to zero, only converges to 0.03.
+        np.testing.assert_allclose(Bn, 0, atol=0.03)
```

Comment:
@f0uriest Can you push the axissymetric bugfix to the anisotropy branch? Then we can merge it to this branch. Maybe that will fix the issue.

Here you can see above that `Phi` is computed correctly, but the result does not converge to zero. This is just applying the shielding theorem.

Reply:
It didn't change anything, but thank you.